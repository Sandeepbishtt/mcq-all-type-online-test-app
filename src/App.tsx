import React from 'react';
import StartPage from './Components/StartPage'
import {BrowserRouter as Router , Routes ,Route} from 'react-router-dom'
import Question from './Pages/Question'

function App() {
  return (
    <Router>
    <h1 style={{textAlign:'center'}}> Online MCQ App </h1>
    <Routes>
 <Route path="/" element ={<StartPage/>}  />
 <Route path="/Question/:id" element ={<Question/>}  />
    </Routes>
    </Router>
  );
}

export default App;
   