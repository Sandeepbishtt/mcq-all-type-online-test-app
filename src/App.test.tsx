import React from 'react';
import { render, screen } from '@testing-library/react';
import App from './App';
import {Provider} from 'react-redux'
import store from './Redux/Store'
import ReactDOM from 'react-dom'

test('renders app component', () => {
  render(
    
      <Provider store={store} >
      <App />
       </Provider>
     
    );
  const linkElement = screen.getByText(/Online MCQ App/i);
  expect(linkElement).toBeInTheDocument();
});

describe('Login component tests', () => {
    let container: HTMLDivElement

    beforeEach(() => {
        container = document.createElement('div');
        document.body.appendChild(container);
        ReactDOM.render(
        <React.StrictMode>
            
            <Provider store={store} >
              <App />
              </Provider>
             
            </React.StrictMode>, container);
    })

    afterEach(() => {
        document.body.removeChild(container);
        container.remove();
    })
    it('check snapshot',()=>{
      expect(container).toMatchSnapshot();
    })

  })