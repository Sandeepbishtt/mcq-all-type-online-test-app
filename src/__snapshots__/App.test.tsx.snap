// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login component tests check snapshot 1`] = `
<div>
  <h1
    style="text-align: center;"
  >
     Online MCQ App 
  </h1>
  <div
    class="MuiBox-root css-qpb54j"
  >
    <form
      data-test="login-form"
      style="margin-left: 1.1rem; margin-top: 2rem;"
    >
      <div
        class="MuiInput-root MuiInput-underline MuiInputBase-root MuiInputBase-colorPrimary form-control css-a88p61-MuiInputBase-root-MuiInput-root"
        required=""
        style="width: 25rem; margin-bottom: 1rem;"
      >
        <input
          class="MuiInput-input MuiInputBase-input css-1x51dt5-MuiInputBase-input-MuiInput-input"
          name="username"
          placeholder="Name"
          required=""
          type="text"
          value=""
        />
      </div>
      <label
        class="MuiInputLabel-root MuiInputLabel-animated MuiFormLabel-root MuiFormLabel-colorPrimary css-9npbnl-MuiFormLabel-root-MuiInputLabel-root"
        id="demo-simple-select-label"
      >
        Gender
      </label>
      <div
        class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
        required=""
        style="width: 25rem; margin-bottom: 1rem;"
      >
        <div
          aria-expanded="false"
          aria-haspopup="listbox"
          aria-labelledby="demo-simple-select-label demo-simple-select"
          class="MuiSelect-select MuiSelect-outlined MuiOutlinedInput-input MuiInputBase-input css-11u53oe-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
          id="demo-simple-select"
          role="button"
          tabindex="0"
        >
          <span
            class="notranslate"
          >
            ​
          </span>
        </div>
        <input
          aria-hidden="true"
          class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
          required=""
          tabindex="-1"
          value=""
        />
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-hfutr2-MuiSvgIcon-root-MuiSelect-icon"
          data-testid="ArrowDropDownIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M7 10l5 5 5-5z"
          />
        </svg>
        <fieldset
          aria-hidden="true"
          class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
        >
          <legend
            class="css-173wfxe"
          >
            <span>
              Gender
            </span>
          </legend>
        </fieldset>
      </div>
      <label
        class="MuiInputLabel-root MuiInputLabel-animated MuiFormLabel-root MuiFormLabel-colorPrimary css-9npbnl-MuiFormLabel-root-MuiInputLabel-root"
        id="demo-simple-select-label"
      >
        Language
      </label>
      <div
        class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
        required=""
        style="width: 25rem; margin-bottom: 1rem;"
      >
        <div
          aria-expanded="false"
          aria-haspopup="listbox"
          aria-labelledby="demo-simple-select-label demo-simple-select"
          class="MuiSelect-select MuiSelect-outlined MuiOutlinedInput-input MuiInputBase-input css-11u53oe-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
          id="demo-simple-select"
          role="button"
          tabindex="0"
        >
          <span
            class="notranslate"
          >
            ​
          </span>
        </div>
        <input
          aria-hidden="true"
          class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
          required=""
          tabindex="-1"
          value=""
        />
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-hfutr2-MuiSvgIcon-root-MuiSelect-icon"
          data-testid="ArrowDropDownIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M7 10l5 5 5-5z"
          />
        </svg>
        <fieldset
          aria-hidden="true"
          class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
        >
          <legend
            class="css-173wfxe"
          >
            <span>
              language
            </span>
          </legend>
        </fieldset>
      </div>
      <button
        class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root  css-sghohy-MuiButtonBase-root-MuiButton-root"
        data-testid="button"
        style="margin-top: 1rem; margin-left: 9rem;"
        tabindex="0"
        type="submit"
      >
        Start MCQ
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </form>
  </div>
</div>
`;
